numpy/core/arrayprint.py

```diff
@@ -401,21 +415,25 @@ def str_format(x):
     return str(x)
 
 def _get_formatdict(data, *, precision, floatmode, suppress, sign, legacy,
-                    formatter, **kwargs):
+                    formatter, exp_format, trim, **kwargs):
     # note: extra arguments in kwargs are ignored
 
     # wrapped in lambdas to avoid taking a code path with the wrong type of data
     formatdict = {
         'bool': lambda: BoolFormat(data),
         'int': lambda: IntegerFormat(data),
         'float': lambda: FloatingFormat(
-            data, precision, floatmode, suppress, sign, legacy=legacy),
+            data, precision, floatmode, suppress, sign, exp_format=exp_format, trim=trim, 
+            legacy=legacy,),
```

Comment:
This line and the similar lines below should be reformatted to shorten the line length
```suggestion
            data, precision, floatmode, suppress, sign, exp_format=exp_format,
            trim=trim, legacy=legacy,),
```